src/account.ts

```diff
@@ -4,154 +4,198 @@ import {
+        }
+        this.client = args.client
     }
 
     get accountName() {
-        return Name.from(this.account_data.account_name)
+        return Name.from(this.data.account_name)
     }
 
-    getPermission(permissionName: NameType): Permission {
-        const permissionObject = this.account_data.permissions.find((permission) =>
+    get systemToken() {
+        return Asset.Symbol.from(this.data.total_resources.cpu_weight.symbol)
+    }
+
+    balance(contract: NameType = 'eosio.token', symbol?: Asset.SymbolType): Promise<Asset> {
+        return new Promise((resolve, reject) => {
+            this.client.v1.chain
+                .get_currency_balance(contract, String(this.accountName), symbol && String(symbol))
+                .then((balances) => {
+                    const balance = (balances as any)[0]
+
+                    if (!balance) {
+                        reject(
+                            new Error(
+                                `No balance found for ${symbol} token of ${contract} contract.`
+                            )
+                        )
+                    }
+
+                    resolve(balance)
+                })
+                .catch((err) => {
+                    if (
+                        err.message.includes('No data') ||
+                        err.message.includes('Account Query Exception')
+                    ) {
+                        reject(new Error(`Token contract ${contract} does not exist.`))
+                    }
+                    reject(err)
+                })
+        })
+    }
+
+    permission(permissionName: NameType): Permission {
+        const permission = this.data.permissions.find((permission) =>
             permission.perm_name.equals(permissionName)
         )
 
-        if (!permissionObject) {
+        if (!permission) {
             throw new Error(
                 `Permission ${permissionName} does not exist on account ${this.accountName}.`
             )
         }
 
-        return new Permission(permissionName, {
-            account: this.accountName,
-            parent: permissionObject.parent,
-            permission: permissionObject.perm_name,
-            auth: Authority.from(permissionObject.required_auth),
-            authorized_by: '............1',
-        })
+        return Permission.from(permission)
     }
 
-    updatePermission(permission: Permission): Action {
-        return this.eosioContract.action('updateauth', permission.actionData)
+    resource(resourceType: ResourceType): Resource {
+        return new Resource(resourceType, this.data)
     }
 
-    removePermission(permissionName: NameType): Action {
-        return this.eosioContract.action('deleteauth', {
-            account: '............1',
-            authorized_by: '............1',
-            permission: permissionName,
+    // TODO: Refactor once resources library is updated
+    resources(sampleAccount?: NameType) {
```

Comment:
What is the `sampleAccount` here? 🤔 Should this be the current account?

Reply:
It's a frequently used account on the network, where the resource values are constantly refreshing due to usage. For that reason it's not great to have it be the users account. 

Kinda a hacky way to calculate this stuff, and an optional value for the resources library. I'm hoping most people won't have to use it or we'll find a way to get rid of it all together.